src/Host/Broker/Impl/Sessions/SessionManager.cs

```diff
@@ -59,6 +60,41 @@ ILogger<Process> hostOutputLogger
             }
         }
 
+        public void CloseAndBlockSessionsCreationForUser(IIdentity user) {
```

Comment:
It might be better to make this return an `IDisposable`, such that disposing it would unblock, to enforce the pairing. And hide `Unblock` as a public method completely.

Also, perhaps make it optional to kill exising sessions? Something like `BlockSessionCreationForUser(..., bool terminateExisting)`?